src/network/mod.rs

```diff
@@ -0,0 +1,154 @@
+mod smol_executor;
+
+use std::net::TcpListener;
+
+use crate::structs::{OutputValues, Clients, Arm};
+
+use self::smol_executor::*;
+use async_compat::*;
+use futures_util::sink::SinkExt;
+use smol::{prelude::*, Async};
+
+use async_tungstenite::{
+    tungstenite::{handshake::derive_accept_key, protocol::Role, Message}, 
+    WebSocketStream
+};
+
+use eyre::{Result, Error};
+use hyper::{
+    Server, service::{make_service_fn, service_fn}, 
+    Request, Body, Response, StatusCode, 
+    header::{
+        HeaderValue, 
+        CONNECTION, UPGRADE, SEC_WEBSOCKET_ACCEPT, SEC_WEBSOCKET_KEY}
+};
+
+
+pub async fn handle_clients(values: Arm<OutputValues>, clients: Clients) {
+    let _span = tracy_client::span!("handle clients");
+    let mut clients = clients.lock().unwrap();
+
+    let values_lock = values.lock().unwrap();
+    let serialized_values = serde_json::to_string(
+        &(*values_lock)
+    ).unwrap(); // Serialization not gonna fail in any possible way
+
+    drop(values_lock); // Droping lock manually just in case
+
+    clients.retain_mut(|websocket| {
+        smol::block_on(async {
+            let next_future = websocket.next();
+
+            let msg_future = 
+                smol::future::poll_once(next_future);
+
+            let msg = match msg_future.await {
+                Some(v) => {
+                    match v {
+                        Some(Ok(v)) => Some(v),
+                        Some(Err(_)) => return false,
+                        None => None,
+                    }
+                },
+                None => None,
+            };
```

Comment:
`match` works nicely on nested patterns too to avoid unnecessary indentation
```rs
let msg = match msg_future.await {
    Some(Some(Ok(v))) => Some(v),
    Some(Some(Err(_))) => return false,
    Some(None) | None => None,
};
```